protocol/statetracker/pairing_updater.go

```diff
@@ -85,6 +90,18 @@ func (pu *PairingUpdater) Update(latestBlock int64) {
 	pu.nextBlockForUpdate = nextBlockForUpdateMin
 }
 
+// updateConsumerSessionManagerCULimits for new virtual epoch when emergency mode is enabled
+func (pu *PairingUpdater) updateConsumerSessionManagerCULimits(virtualEpoch uint64) {
+	for _, consumerSessionManagers := range pu.consumerSessionManagersMap {
+		for _, consumerSessionManager := range consumerSessionManagers {
+			if consumerSessionManager == nil {
+				continue
+			}
+			consumerSessionManager.UpdateMaxCULimit(virtualEpoch, pu.currentVirtualEpoch)
```

Comment:
should implement update virtual epoch on consumer sessions manager instead of directly calling the max CU function, for modularity, in case we need to add more stuff to virtual epoch changes, also lower coupling with pairing updater and consumer sessions manager